### DISCUSSION TITLE
Move the client-side code of the prefetch to the end of the `</body>` instead of the beginning

### DISCUSSION THREAD

**Discussion by kricsleo** [withastro/roadmap#761] [Proposal]
# Summary

Move the client-side code of the prefetch to the end of the `</body>` instead of the beginning.

I think this is a minor optimisation.

# Background & Motivation

Putting less high priority js code at the end of the body can speed up the rendering of the page.

There's already a related task on modifying prefetch, but there doesn't seem to be any mention of it in there, so I'm making this suggestion, and if there's anything inappropriate please just point it out!

related: https://github.com/withastro/astro/pull/8951

Maybe we could make some changes to this:

https://github.com/withastro/astro/blob/3988bbcc9ead0b9af60b8a8749a0ad25c686bde3/packages/integrations/prefetch/src/index.ts#L10-L14

For example, support for `'body-end'` in `injectScript`

```ts
injectScript('body-end', /** prefetch code goes here */);
```

# Goals

Let's go faster even more! 🚀

**Comment by Princesseuh** [comment]
The script gets added as `type="module"`, which means it's non-blocking and is only executed after the page is ready. Putting it at the end wouldn't change much (if anything)

![image](https://github.com/withastro/roadmap/assets/3019731/74a0bb36-a66e-4c43-b88b-1936c0f83fcf)

**Reply by kricsleo** [reply]
Oh, that's something I've picked up today! Thank you :)

Also attach a reference link for someone who got here too: https://v8.dev/features/modules#defer
